coordinator/src/node.rs

```diff
@@ -51,90 +142,162 @@ impl Node {
             accept_collateral: margin_coordinator,
             fee_rate: 2,
             contract_infos: vec![ContractInputInfo {
-                contract_descriptor: dummy_contract_descriptor(total_collateral),
+                contract_descriptor,
                 oracles: OracleInput {
-                    public_keys: vec![trade_params.oracle_pk],
+                    public_keys: vec![self.inner.oracle_pk()],
                     event_id: format!("{contract_symbol}{maturity_time}"),
                     threshold: 1,
                 },
             }],
         };
 
+        let channel_details = self.get_counterparty_channel(trade_params.pubkey)?;
+        self.inner
+            .propose_dlc_channel(&channel_details, &contract_input)
+            .await?;
+        Ok(())
+    }
+
+    fn get_counterparty_channel(&self, trader_pubkey: PublicKey) -> Result<ChannelDetails> {
         let channel_details = self.inner.list_usable_channels();
         let channel_details = channel_details
-            .iter()
-            .find(|c| c.counterparty.node_id == trade_params.pubkey)
-            .context("Channel details not found")?;
+            .into_iter()
+            .find(|c| c.counterparty.node_id == trader_pubkey)
+            .context("Channel details not found")
+            .map_err(|e| anyhow!("{e:#}"))?;
+        Ok(channel_details)
+    }
+}
 
-        self.inner
-            .propose_dlc_channel(channel_details, &contract_input)
-            .await?;
+fn get_margins(trade_params: &TradeParams) -> (u64, u64) {
+    let margin_coordinator = calculate_margin(
+        trade_params.execution_price,
+        trade_params.quantity,
+        COORDINATOR_LEVERAGE,
+    );
+    let margin_trader = calculate_margin(
+        trade_params.execution_price,
+        trade_params.quantity,
+        trade_params.leverage,
+    );
 
-        Ok(())
+    (margin_coordinator, margin_trader)
+}
+
+fn get_leverages(trade_params: &TradeParams) -> (f64, f64) {
```

Comment:
🙃 To avoid confusion when reading the code it would be good if we could stick to always using the same order when returning tuples; in ItchySats we agreed on `(long, short)` (long was always first, short second).
Here we use `(short, long)` - we might want to turn this around.

Reply:
I adapted it to always be `long, short`